dvc/stage.py

```diff
@@ -241,6 +241,20 @@ def __repr__(self):
             path=self.relpath if self.path else "No path"
         )
 
+    def __hash__(self):
+        return hash(self.path_in_repo)
+
+    def __eq__(self, other):
+        return (
+            self.__class__ == other.__class__
+            and self.repo == other.repo
```

Comment:
do we have a "semantic" `__eq__` for the `repo` class?

Reply:
No, I meant the same repo here. May just change it to `is` to make the intention clear.